staging/src/k8s.io/client-go/discovery/discovery_client_test.go

```diff
@@ -2833,6 +2834,41 @@ func TestUseLegacyDiscovery(t *testing.T) {
 	client.ServerGroups()
 }
 
+func TestIsGroupDiscoveryFailedError(t *testing.T) {
+	t.Parallel()
+
+	for _, tc := range []struct {
+		name string
+		err  error
+		want bool
+	}{
+		{
+			name: "NoGroupDiscoveryFailedError",
+			err:  errors.New("foo"),
+			want: false,
+		},
+		{
+			name: "GroupDiscoveryFailedError",
+			err:  &ErrGroupDiscoveryFailed{},
+			want: true,
+		},
+		{
+			name: "WrappedGroupDiscoveryFailedError",
+			err:  fmt.Errorf("foo: %w", &ErrGroupDiscoveryFailed{}),
+			want: true,
+		},
+	} {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			got := IsGroupDiscoveryFailedError(tc.err)
```

Comment:
Could you please add a test case for `err` is nil?

Reply:
Yep, done :+1: